packages/babel-parser/src/types.js

```diff
@@ -541,6 +540,8 @@ export type ConditionalExpression = NodeBase & {
 export type CallOrNewBase = NodeBase & {
   callee: Expression | Super | Import,
   arguments: Array<Expression | SpreadElement>, // TODO: $ReadOnlyArray
+  optional?: boolean, // For OptionalCallExpression only
```

Comment:
Is moving this something we can avoid if we refactored the code to verify the type before setting the flag? Even if it's a pain, it seems like ideally we'd keep the types as specific as possible.

Reply:
Agreed on keeping types specific.

I went down a bit of a rabbit hold with refactoring last night... but actually on second look, seems like simply removing the CallExpression typecast makes flow happy.